p2p/discover/v4_udp.go

```diff
@@ -162,31 +174,16 @@ func nodeToRPC(n *node) rpcNode {
 	return rpcNode{ID: ekey, IP: n.IP(), UDP: uint16(n.UDP()), TCP: uint16(n.TCP())}
 }
 
-// packet is implemented by all protocol messages.
-type packet interface {
-	// preverify checks whether the packet is valid and should be handled at all.
-	preverify(t *udp, from *net.UDPAddr, fromID enode.ID, fromKey encPubkey) error
-	// handle handles the packet.
-	handle(t *udp, from *net.UDPAddr, fromID enode.ID, mac []byte)
-	// name returns the name of the packet for logging purposes.
-	name() string
-}
-
-type conn interface {
-	ReadFromUDP(b []byte) (n int, addr *net.UDPAddr, err error)
-	WriteToUDP(b []byte, addr *net.UDPAddr) (n int, err error)
-	Close() error
-	LocalAddr() net.Addr
-}
-
-// udp implements the discovery v4 UDP wire protocol.
-type udp struct {
-	conn        conn
+// UDPv4 implements the v4 wire protocol.
+type UDPv4 struct {
```

Comment:
The UDP naming is somewhat confusing. Consider naming this to something that makes it more explicit that this is the discovery protocol as opposed to a version of UDP (since discovery is a layer atop UDP). Some examples include `PeerDiscoveryManagerv4`, `PeerDiscovererV4`, `DiscoveryV4`.

Reply:
The discovery part of the name is in the package, i.e. the full name of this type is `discover.UDPv4`. I'm about to add `discover.UDPv5`.

Reply:
Yes, my bad about adding stutter. The confusion over having UDP in the name is the main point. I'm also not sure if the lowercase `v` is idiomatic golang. If it were me I would go with `discover.V4Manager` or `discover.ManagerV4`.